orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -92,6 +116,30 @@ class ORTTrainerOptions(object):
                         'enable_adasum' : {
                             'type' : 'boolean',
                             'default' : False
+                        },
+                        'sliced_schema' : {
+                            'type': 'dict',
+                            'default' : {},
+                            'required': False,
+                            # the keys in `sliced_schema` are strings.
+                            'keysrules': {'type': 'string'},
+                            # the values in `sliced_schema` are lists of integers.
+                            'valuesrules': {
+                                'type': 'list',
+                                'schema': {'type': 'integer'}
+                            }
+                        },
+                        'sliced_axes' : {
+                            'type': 'dict',
+                            'keysrules': {'type': 'string'},
+                            'valuesrules': {
+                                'type': 'integer',
+                                'default': 0
+                            }
+                        },
+                        'sliced_tensor_names' : {
+                            'type': 'list',
+                            'schema': {'type': 'string'}
```

Comment:
I asm not sure if you need extra logic to validate this, but if you do, you can implement a custom validator for that. 

Refer to how model_desc is validated using cerberus at https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/model_desc_validation.py.

Note this file has a lot more than just schema validation. You just need the schema validation part, which starts with `MODEL_DESC_SCHEMA` being defined with a list which has a schema with a custom validator that implements the kind of logic you need

`def _model_desc_inputs_validation(field, value, error):` is employed and have this fixed arguments you must use

Then you just validate the schema as usual:

```
        validator = cerberus.Validator(MODEL_DESC_SCHEMA)
        self._validated = validator.validated(self._validated)
        if self._validated is None:
            raise ValueError(f'Invalid model_desc: {validator.errors}')
```

Reply:
Looks like we don't an extra validator.

---
In reply to: [542797569](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r542797569) [](ancestors = 542797569)